validate/validate.h

```diff
@@ -42,25 +43,44 @@ class UnimplementedException : public std::runtime_error {
 using ValidationMsg = std::string;
 
 class BaseValidator {
+public:
+  static bool AbstractCheckMessage(const google::protobuf::Message& m, ValidationMsg* err) {
+    // Polymorphic lookup is used to see if there is a matching concrete validator. If so, call it.
+    // Otherwise return success.
+    auto it = abstractValidators().find(std::type_index(typeid(m)));
+    if (it == abstractValidators().end()) {
+      return true;
+    }
+    return it->second(m, err);
+  }
+
 protected:
   static std::unordered_map<std::type_index, BaseValidator*>& validators() {
     static auto* validator_map = new std::unordered_map<std::type_index, BaseValidator*>();
     return *validator_map;
   }
+
+  static std::unordered_map<std::type_index,
+                            std::function<bool(const google::protobuf::Message&, ValidationMsg*)>>&
+  abstractValidators() {
```

Comment:
Can you add some more detailed comments on the distinction between abstract and non-abstract validators here? I.e. why we have both.